lib/src/models/version_compatibility.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/cardano_version.dart';
+
+part 'version_compatibility.freezed.dart';
+
+@freezed
+class VersionCompatibility with _$VersionCompatibility {
+  const factory VersionCompatibility({
+    required bool isCompatible,
+    String? recommendedVersion,
+    required bool supportsNativeScriptHashDerivation,
+  }) = _VersionCompatibility;
+
+  static VersionCompatibility checkVersionCompatibility(CardanoVersion version) {
```

Comment:
static functions returning an instance of the class should generally be factories instead. So please change this to:
```dart
cont factory VersionCompatibility.checkVersionCompatibility(CardanoVersion version) {
```

Reply:
done